src/apps/invitations/services.py

```diff
@@ -124,6 +130,50 @@ async def send_invitation(self, schema: InvitationRequest) -> Invitation:
 
         return invitation
 
+    async def _validate_invitation(
+        self, invitation_request: InvitationRequest
+    ):
+        is_exist = await AppletService().exist_by_id(
+            invitation_request.applet_id
+        )
+        if not is_exist:
+            raise AppletDoesNotExist(
+                f"Applet by id {invitation_request.applet_id} does not exist."
+            )
+
+        access_service = UserAppletAccessService(
+            self._user.id, invitation_request.applet_id
+        )
+        if invitation_request.role == Role.RESPONDENT:
+            role = await access_service.is_respondents_manager()
+        elif invitation_request.role in [
+            Role.MANAGER,
+            Role.COORDINATOR,
+            Role.EDITOR,
+            Role.REVIEWER,
+        ]:
+            role = await access_service.is_organizer()
+        else:
+            # Wrong role to invite
+            raise DoesNotHaveAccess(
+                message="You can not invite user with "
+                f"role {invitation_request.role.name}."
+            )
+
+        if not role:
+            # Does not have access to send invitation
+            raise DoesNotHaveAccess(
+                message="You do not have access to send invitation."
+            )
+        elif RoleLevel.by_role(Role(role)) > RoleLevel.by_role(
+            Role(invitation_request.role)
+        ):
+            # TODO: remove this validation if it is not needed
+            # Can not invite users by roles where own role level is lower.
+            raise DoesNotHaveAccess(
+                message="You do not have access to send invitation."
+            )
```

Comment:
This construction is overcomplicated: `RoleLevel.by_role(Role(role)) > RoleLevel.by_role(Role(invitation_request.role))`

Why you are converting the role you get from the `access_service.is_orginizer()` to the Role on line 168?

First of all, you can pass this argument as it is instead of transforming it again.